automatic str() with sum type of []primitive causes C code gen error

**V version:** 

```
V full version: V 0.2.2 6171e12.51075ff
```

**OS:** 

```
OS: macos, macOS, 11.4, 20F71
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)
```

**What did you do?**

Run `v test /tmp/tmp_test.v`

```
type Ints = []int
type Strings = []string
type Result = Ints | Strings

fn test_foo() {
	res := Result(Ints([1, 2, 3]))
	println(res)
}
```

**What did you expect to see?**

No C error
 
**What did you see instead?**

```
---- Testing... ----------------------------------------------------------------
Summary for all V _test.v files: 1 failed, 1 total. Runtime: 398 ms.

FAIL   398.060 ms /private/tmp/tmp_test.v
==================
/tmp/v/test_session_88824/tmp_test.18416934872650190305.tmp.c:2510:19: error: expected expression
        string tmp_ds = []string_str(it);
                         ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

--------------------------------------------------------------------------------
```

--- a/vlib/v/gen/c/auto_str_methods.v
+++ b/vlib/v/gen/c/auto_str_methods.v
@@ -221,12 +221,7 @@ fn (mut g Gen) gen_str_for_option(typ ast.Type, styp string, str_fn_name string)
 }
 
 fn (mut g Gen) gen_str_for_alias(info ast.Alias, styp string, str_fn_name string) {
-	sym := g.table.get_type_symbol(info.parent_type)
-	sym_has_str_method, _, _ := sym.str_method_info()
-	mut parent_str_fn_name := styp_to_str_fn_name(sym.name.replace('.', '__'))
-	if !sym_has_str_method {
-		parent_str_fn_name = g.gen_str_for_type(info.parent_type)
-	}
+	parent_str_fn_name := g.gen_str_for_type(info.parent_type)
 	mut clean_type_v_type_name := util.strip_main_name(styp.replace('__', '.'))
 	g.type_definitions.writeln('static string ${str_fn_name}($styp it); // auto')
 	g.auto_str_funcs.writeln('static string ${str_fn_name}($styp it) { return indent_${str_fn_name}(it, 0); }')

--- a/vlib/v/tests/string_interpolation_sumtype_test.v
+++ b/vlib/v/tests/string_interpolation_sumtype_test.v
@@ -1,0 +1,9 @@
+type Ints = []int
+type Strings = []string
+type Result = Ints | Strings
+
+fn test_string_interpolation_sumtype() {
+	res := Result(Ints([1, 2, 3]))
+	println(res)
+	assert '$res' == 'Result(Ints([1, 2, 3]))'
+}

